spec/models/tag_wrangling_spec.rb

```diff
@@ -0,0 +1,536 @@
+          expect(fandom.meta_tags.reload).to include(meta)
+          expect(fandom.direct_meta_tags.reload).not_to include(meta)
+        end
+      end
+    end
+
+    describe "adding multiple meta tags" do
+      let(:fandom) { create(:canonical_fandom) }
+      let(:meta1) { create(:canonical_fandom) }
+      let(:meta2) { create(:canonical_fandom) }
+
+      it "adds both direct meta tags" do
+        fandom.update!(meta_tag_string: "#{meta1.name}, #{meta2.name}")
+        expect(fandom.meta_tags.reload).to include(meta1)
+        expect(fandom.meta_tags.reload).to include(meta2)
+        expect(fandom.direct_meta_tags.reload).to include(meta1)
+        expect(fandom.direct_meta_tags.reload).to include(meta2)
+      end
+
+      it "updates inheritance for both meta tags" do
+        meta_meta1 = create(:canonical_fandom)
+        meta_meta1.sub_tags << meta1
+        meta_meta2 = create(:canonical_fandom)
+        meta_meta2.sub_tags << meta2
+
+        fandom.update!(meta_tag_string: "#{meta1.name}, #{meta2.name}")
+        expect(fandom.meta_tags.reload).to include(meta_meta1)
+        expect(fandom.meta_tags.reload).to include(meta_meta2)
+      end
+    end
+
+    describe "adding a meta tag and a sub tag simultaneously" do
+      let(:fandom) { create(:canonical_fandom) }
+      let(:meta) { create(:canonical_fandom) }
+      let(:sub) { create(:canonical_fandom) }
+
+      it "adds both direct meta taggings" do
+        fandom.update!(meta_tag_string: meta.name, sub_tag_string: sub.name)
+        expect(fandom.meta_tags.reload).to include(meta)
+        expect(fandom.direct_meta_tags.reload).to include(meta)
+        expect(fandom.sub_tags.reload).to include(sub)
+        expect(fandom.direct_sub_tags.reload).to include(sub)
+      end
+
+      it "adds the inherited meta tagging" do
+        fandom.update!(meta_tag_string: meta.name, sub_tag_string: sub.name)
+        expect(sub.meta_tags.reload).to include(meta)
+        expect(sub.direct_meta_tags.reload).not_to include(meta)
+      end
+    end
+  end
+
+  describe "when the resque queue is long" do
+    # https://otwarchive.atlassian.net/browse/AO3-4077
+    it "deletes the inherited meta tag when both direct ones are removed" do
+      lowest = create(:canonical_fandom)
+      middle = create(:canonical_fandom)
+      highest = create(:canonical_fandom)
+      middle.meta_tags << highest
+      middle.sub_tags << lowest
+
+      suspend_resque_workers do
+        middle.reload
+        middle.update!(associations_to_remove: [lowest.id, highest.id])
+      end
+
+      expect(lowest.meta_tags.reload).not_to include(highest)
+    end
+
+    it "deletes meta filters when a syn and a meta tag are deleted" do
+      canonical = create(:canonical_fandom)
+      meta = create(:canonical_fandom)
+      canonical.meta_tags << meta
+      synonym = create(:fandom, merger: canonical)
+
+      work = create(:work, fandom_string: synonym.name)
+
+      suspend_resque_workers do
```

Comment:
Is it possible for this block to miss any `Resque.enqueue` calls before it? Should all the set up steps be inside the block too?

Reply:
It is guaranteed to miss any `Resque.enqueue` calls before it, but that's by design -- the point of the test is to have everything working properly before the wrangling queue is delayed and breaks (or preferably doesn't break) everything.